markdown/opentelemetry-intro/opentelemetry-intro.md

```diff
@@ -0,0 +1,319 @@
+ 
+
+		trace.set_tracer_provider(TracerProvider())
+		jaeger_exporter = jaeger.JaegerSpanExporter(
+			service_name="<Boomi> Listen for Salesforce Platform Account, publish Solace DriverUpserted",
+			agent_host_name="localhost",
+			agent_port=6831,
+		)
+1. Now that there is connectivity to Jaeger, the code specifies that the spans we create will be send in batches using the BatchExportSpanProcessor.
+
+		trace.get_tracer_provider().add_span_processor(
+		BatchExportSpanProcessor(jaeger_exporter)
+		
+
+1. Using the trace factory that you created above, you get an instance of a trace, called tracer.  Using tracer, you create a span, which describes what happens in this microservice.
+
+		tracer = trace.get_tracer(__name__)
+		# THIS IS PER https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+		parentSpan = tracer.start_span(
+			"RideUpdated send",
+			kind=SpanKind.PRODUCER,
+			attributes={
+				"messaging.system": "solace",
+				"messaging.destination": "RideUpdated send",
+				"messaging.destination-kind": "topic",
+				"messaging.protocol": "jcsmp",
+				"messaging.protocol_version": "1.0",
+				"messaging.url": "url of solace box"}
+		)
+
+1. Now that the span is established, you need to get the span's trace_id and the span_id 
+
+		trace_id = parentSpan.get_context().trace_id
+		span_id = parentSpan.get_context().span_id`
+
+1. As the script publishes the event, it includes the trace id and the span id as two headers in the message.  Downstream consumers now have access to them.
+
+		outbound_msg = OutboundMessage.builder() \
+			.with_property("trace_id", str(trace_id)) \
+			.with_property("span_id", str(span_id)) \
+			.build("Hello World! This is a message published from Python!")
+ 
+1. Finally, end the parentSpan, so it will be sent to Jaeger
+
+		parentSpan.end()
+
+## Create the message consumers
+
+Duration: 0:10:00
+
+Now that "Salesforce" is publishing an event to Solace, you need to get the "REST" consumer and the "database" consumers up and running. It's basically the same procedure as before.
+
+1. In IDLE, open `solace_telemetry_consumer_Database.py`
+1. Update the connection parameters to match your Solace event broker:
+
+		hostName =  "<SMF Host copied from Solace Cloud>"
+		vpnName = "<Message VPN copied from Solace Cloud>"
+		solaceUserName =  "solace-cloud-client"
+		solacePassword = "<Password copied from Solace Cloud>"
+1. Click on  Run > Run Module ![idle_run](img/idle_run.png "idle_run")
+1. You should see something like:
+
+		2020-08-26 08:32:20,449 [INFO] pysolace.messaging.core: [solace_session.py:470]  ESTABLISH SESSION ON HOST [tcp://mr-d8f4yze27kt.messaging.solace.cloud:55555]
+		Execute Direct Consume - String
+		Subscribed to: opentelemetry/helloworld
+
+1. The consumer listens for incoming message for 120 seconds, and then stops.  At that point you'll see a message that says:
```

Comment:
With the new changes, well change that to "The consumer listens for incoming message until the user interrupts it"